internal/ingress/controller/nginx.go

```diff
@@ -660,16 +672,64 @@ func (n *NGINXController) OnUpdate(ingressCfg ingress.Configuration) error {
 		}
 	}
 
-	err = ioutil.WriteFile(cfgPath, content, 0644)
-	if err != nil {
-		return err
-	}
+	if n.cfg.DynamicReload {
+		src, _ := ioutil.ReadFile(cfgPath)
+		if !bytes.Equal(src, content) {
+			err = ioutil.WriteFile(cfgPath, content, 0644)
+			if err != nil {
+				return err
+			}
 
-	o, err := exec.Command(n.binary, "-s", "reload", "-c", cfgPath).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("%v\n%v", err, string(o))
-	}
+			o, err := exec.Command(n.binary, "-s", "reload", "-c", cfgPath).CombinedOutput()
+			if err != nil {
+				return fmt.Errorf("%v\n%v", err, string(o))
+			}
+		} else {
+			glog.Infof("NGINX reload not needed, executing live update only\n")
+		}
+
+		content, err = n.td.Write(tc)
+
+		if err != nil {
+			return err
+		}
 
+		retries := 1
+		updateOK := false
+		for retries <= 5 {
+			if !updateOK {
+				if retries > 1 {
+					glog.Infof("NGINX dynamic update (retrying)\n")
+				}
+				resp, err := http.Post("http://localhost:"+strconv.Itoa(n.cfg.ListenPorts.Status)+"/nginx_update",
+					"application/json", bytes.NewBuffer(content))
+				if err != nil {
+					time.Sleep(1 * time.Second)
+					glog.Infof("NGINX dynamic update not ready\n")
+				} else if resp.StatusCode != 200 {
+					time.Sleep(1 * time.Second)
+					glog.Infof("NGINX dynamic update not ready\n")
+				} else {
+					updateOK = true
+					glog.Infof("NGINX dynamic update OK\n")
```

Comment:
can you add something like 
```go
if glog.V(2) {
  glog.Infof("Updating nginx endpoints with\n%v\n",content)
}
```

so with can see the content used by lua

Reply:
done